S3/Config.py

```diff
@@ -324,14 +326,15 @@ def role_config(self):
                 web_identity_token_file = os.environ.get('AWS_WEB_IDENTITY_TOKEN_FILE')
                 if web_identity_token_file:
                     with open(web_identity_token_file) as f:
-                        web_identity_token = f.read()
+                        web_identity_token = f.read().rstrip()
                     params['Action'] = 'AssumeRoleWithWebIdentity'
                     params['WebIdentityToken'] = web_identity_token
                 encoded_params = '&'.join([
                     '%s=%s' % (k, s3_quote(v, unicode_output=True))
                     for k, v in params.items()
                 ])
-                sts_endpoint = "sts.amazonaws.com"
+                if len(sts_endpoint) == 0:
+                    sts_endpoint = os.environ.get("AWS_STS_ENDPOINT", default="sts.amazonaws.com") 
```

Comment:
In fact, usually "environment" vars have the higher priority to override the config, so it would be better that it comes first.
So, probably, if i'm not wrong, you can put "sts.amazonaws.com" directly as the default config in the config file.
So then you will not need your "if" anymore:
`sts_endpoint = os.environ.get("AWS_STS_ENDPOINT", sts_endpoint)`

(To be noted, you don't need to put the "default=" because it is the single and usual second argument for such getters)